starknet_devnet/starknet_wrapper.py

```diff
@@ -305,11 +316,22 @@ async def deploy(self, deploy_transaction: Deploy) -> Tuple[int, int]:
         try:
             preserved_block_info = self.__update_block_number()
```

Comment:
The block number is updated here, but later in L326 we are using the old block number retrieved in L295. Something is suspicious here.

Reply:
done

Reply:
Let's talk it through on the call